packages/langium/src/grammar/ast-reflection-interpreter.ts

```diff
@@ -0,0 +1,115 @@
+/******************************************************************************
+ * Copyright 2022 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { AstNode, AstReflection, TypeMandatoryField, TypeMetaData } from '../syntax-tree';
+import { isAstNode } from '../utils/ast-util';
+import { MultiMap } from '../utils/collections';
+import { LangiumDocuments } from '../workspace/documents';
+import { Grammar } from './generated/ast';
+import { createLangiumGrammarServices } from './langium-grammar-module';
+import { collectAst } from './type-system/type-collector';
+import { AstTypes, Property } from './type-system/types-util';
+
+let emptyDocuments: LangiumDocuments;
+
+export function interpreteAstReflection(grammar: Grammar, documents?: LangiumDocuments): AstReflection {
+    if (!emptyDocuments && !documents) {
+        emptyDocuments = createLangiumGrammarServices().shared.workspace.LangiumDocuments;
+    }
+    const collectedTypes = collectAst(documents ?? emptyDocuments, [grammar]);
+    const allTypes = collectedTypes.interfaces.map(e => e.name).concat(collectedTypes.unions.map(e => e.name));
+    const references = buildReferenceTypes(collectedTypes);
+    const metaData = buildTypeMetaData(collectedTypes);
+    const superTypeMap = buildSupertypeMap(collectedTypes);
```

Comment:
Question: How are sub types handled in ast reflection? 
For `interface A extends B, C {}` will there be an entry for `A` with all inherited mandatory entries from `B` and `C`?